### PR TITLE
Pip compile multi

### PR DESCRIPTION
**Pull request by mozbhearsum**
This is part 1 of bringing Balrog in line with our Best Practices.

I ended up dropping a bunch unused dependencies as part of this. Largely this is:
* Things that used to be used by scripts (like gcloud-aio, aiohttp, etc.)
* `sphinx`, which arguably should be in `local.in` (I'm having trouble deciding)

...and a ton of stuff they depend on.

I've tested this locally by running `run-tests.sh`, `tox`, and `docker-compose build --no-cache` - which should hit all of the relevant paths. Hopefully Taskcluster passes, too :).

### CODE REVIEW

**Comment by mozbhearsum** [comment]
(Test failure is unrelated, but it has reminded me that I still need to update the agent.)

**Comment by mozbhearsum** [comment]
I also discovered that when I converted `.taskcluster.yml` to v1, I accidentally dropped the client tests. I've added those back, and fixed some minor issues that have cropped up since.

File: scripts/test-rules.py [hunk 1]
```diff
@@ -2,20 +2,14 @@
...
 import os
 import os.path
 import re
 import site
 import time
 
-try:
-    import json
-
-    assert json  # to shut pyflakes up
```
Comment (JohanLorenzo) [inline scripts/test-rules.py]: \o/

File: setup.py [hunk 2]
```diff
@@ -7,9 +7,12 @@
 # of the packages in it. However, setuptools doesn't support parsing this type
 # of file, so we need to strip those out before passing the requirements along
 # to it.
-with open(path.join(here, "requirements.txt")) as f:
+with open(path.join(here, "requirements", "base.txt")) as f:
     requirements = []
     for line in f:
+        # Skip empty lines
```
Comment (JohanLorenzo) [inline setup.py]: Same comments as `agent/setup.py`

File: .taskcluster.yml [hunk 3]
```diff
@@ -112,6 +112,30 @@ tasks:
...
+              maxRunTime: 1200
+              image: "mozillareleases/python-test-runner"
+              env:
+                NO_VOLUME_MOUNT: 1
+              command:
+                - "/bin/bash"
+                - "-c"
+                - "git clone ${repo_url} balrog && cd balrog && git checkout ${head_sha} && cd client && bash run-tests.sh"
+              features:
+                dind: true
```
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply .taskcluster.yml]: They do, actually. We run the tests in Docker: https://github.com/mozilla/balrog/blob/master/client/run-tests.sh

One of the things I'm going to do in part 2 or 3 is change this to build an image in an upstream task, and run the tests out of that.

File: agent/setup.py [hunk 4]
```diff
@@ -9,12 +9,15 @@
 # of the packages in it. However, setuptools doesn't support parsing this type
 # of file, so we need to strip those out before passing the requirements along
 # to it.
-with open(path.join(here, "requirements.txt")) as f:
+with open(path.join(here, "requirements", "base.txt")) as f:
     requirements = []
     for line in f:
+        # Skip empty lines
```
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply agent/setup.py]: Yes, we should! I'm wondering how this is working right now without doing that, actually...

File: .taskcluster.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply .taskcluster.yml]: Oh don't mind me, then :) 👍 to building an upstream image!

**Comment by mozbhearsum** [comment]
Whoops, I just noticed you already approved this with nits. I'm going to go ahead and merge it.

**Pull request merged**